tools/add_triage_signature.py

```diff
@@ -383,8 +384,87 @@ class StorageDetailSignature(Signature):
     def __init__(self, jira_client):
         super().__init__(jira_client, comment_identifying_string="h1. Host storage details:")
 
-    def _update_ticket(self, url, issue_key, should_update=False):
+    class SmartctlExitCode(Flag):
+        """
+        Taken from `man smartctl` "EXIT STATUS" section
+        """
+        COMMANDLINE_DID_NOT_PARSE = auto()
+        DEVICE_OPEN_FAILED = auto()
+        SOME_ATA_COMMAND_FAILED_OR_SMART_CHECKSUM_ERROR = auto()
+        DISK_FAILING = auto()
+        PREFAIL_ATTRIBUTRES_UNDER_THRESHOLD = auto()
+        DISK_OK_SOME_ATTRIBUTES_UNDER_THRESHOLD = auto()
+        LOG_HAS_RECORDS_OF_ERRORS = auto()
+        SELF_TEST_LOG_HAS_RECORDS_OF_ERRORS = auto()
+
+    @classmethod
+    def _parse_smart_internal(cls, smart):
+        output = []
+        smartctl = smart["smartctl"]
+        exit_code = smartctl["exit_status"]
+
+        if exit_code != 0:
+            output.append("non-zero smartctl exit code: " + ", ".join(flag.name for flag in cls.SmartctlExitCode
+                                                                      if flag in cls.SmartctlExitCode(exit_code)))
+            if cls.SmartctlExitCode(exit_code) == cls.SmartctlExitCode.SOME_ATA_COMMAND_FAILED_OR_SMART_CHECKSUM_ERROR:
+                if "smart_status" not in smart:
+                    output.append(
+                        "{color:green}*this typically happens on virtual disks*{color} that don't actually have S.M.A.R.T. information,"
+                        " but may also happen due to other reasons")
```

Comment:
So I wonder if green is the right color in this case
Maybe no color is the right thing here

Reply:
or some other neutral color

Reply:
It's a long sentence, and it generally means "Everything is probably OK, this is probably just a virtual disk, SMART is uninteresting". I want to draw the user to the "this is okay part" to reduce mental effort when going over this long table of disks.